src/screens/Home/index.tsx

```diff
@@ -1,7 +1,195 @@
+      } else {
+        const token = getCookie('token');
+        const data = await api.get('posts', token);
+
+        setPosts(data);
+        sessionStorage.setItem('posts', JSON.stringify(data));
+      }
+    })();
+  }, []);
+
+  useEffect(() => {
+    if (filters.date) {
+      setPosts((prevState) => sortedByDate(prevState));
+    } else if (filters.popular) {
+      setPosts((prevState) => sortedByPopular(prevState));
+    } else if (!filters.date && !filters.popular) {
+      const sessionPosts = sessionStorage.getItem('posts');
+      sessionPosts ? setPosts(JSON.parse(sessionPosts)) : '';
+    }
+  }, [filters.date, filters.popular]);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const mainBlock: HTMLElement | null = document.querySelector('main')!;
+      const currentScrollPos = mainBlock.scrollTop;
+      const isVisible = prevScrollPos > currentScrollPos;
+
+      setPrevScrollPos(currentScrollPos);
+      setVisible(isVisible);
+    };
+
+    document.querySelector('main')!.addEventListener('scroll', handleScroll);
+    return () => {
+      document.querySelector('main')!.removeEventListener('scroll', handleScroll);
+    };
+  }, [prevScrollPos]);
+
+  return (
+    <S.Wrapper>
+      <S.Header className={visible ? '' : 'hidden'}>
+        <S.Filters>
+          <S.IconWrapper
+            ref={menuRef}
+            onClick={() => {
+              setIsOpen((prev) => !prev);
+            }}
+          >
+            <S.FilterIcon className={isOpen ? 'opened' : ''} />
+          </S.IconWrapper>
+          <S.FiltersMenu className={isOpen ? 'opened' : ''}>
+            <S.MenuItem>
+              <S.CheckboxLabel>
+                <S.InputCheck
+                  type='checkbox'
+                  name='date'
+                  checked={filters.date}
+                  onChange={handleFiltersChange}
+                />
+                <S.Span>Date</S.Span>
+              </S.CheckboxLabel>
+            </S.MenuItem>
+            <S.MenuItem>
+              <S.CheckboxLabel>
+                <S.InputCheck
+                  type='checkbox'
+                  name='popular'
+                  checked={filters.popular}
+                  onChange={handleFiltersChange}
+                />
+                <S.Span>Popular</S.Span>
+              </S.CheckboxLabel>
+            </S.MenuItem>
+          </S.FiltersMenu>
+        </S.Filters>
+        <S.SearchInput
+          name='search'
+          value={inputValue}
+          onChange={handleInputChange}
+          placeholder='Search...'
+          type='text'
+        />
+        <S.Tabs className={activeTab === 0 ? 'left' : 'right'}>
+          {tabs.map((tab, index) =>
+            activeTab === index ? (
+              <S.ActiveTab key={index} onClick={() => handleTabClick(index)} to={tab.to}>
+                {tab.title}
+              </S.ActiveTab>
+            ) : (
+              <S.Tab key={index} onClick={() => handleTabClick(index)} to={tab.to}>
```

Comment:
Could you add props for styles like isActive, no needed have 2 separate components